### PR TITLE
WIP - new extension: permalink_copy

### PR DESCRIPTION
**Pull request by manavortex**
Added a new extension to change click behaviour of permalink

### CODE REVIEW

File: Extensions/permalink_copy.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+//* TITLE permalink_copy **//
```
Comment (invalidCards) [inline Extensions/permalink_copy.js:1]:
```suggestion
//* TITLE Permalink Copy **//
```
This title is user-facing

File: Extensions/permalink_copy.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+//* VERSION 1.0.0 **//
+//* DESCRIPTION **//
```
Comment (invalidCards) [inline Extensions/permalink_copy.js:3]: Please add a description

File: Extensions/permalink_copy.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+//* DEVELOPER New-XKit **//
```
Comment (invalidCards) [inline Extensions/permalink_copy.js:4]:
```suggestion
//* DEVELOPER new-xkit **//
```
consistency

File: Extensions/permalink_copy.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+function interceptClickEvent(event) {	 
+	if (shouldRun(event.target)) onPermalinkClick(event);
+}
+
+function interceptMmbClickEvent(event) { 
+	if (shouldRun(event.target)) onPermalinkClick(event);
+}
+
+XKit.extensions.permalink_copy = new Object({
```
Comment (invalidCards) [inline Extensions/permalink_copy.js:65]: Please move the preference definition to the top of the file (under the header) for consistency

File: Extensions/permalink_copy.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+			type: "checkbox",
+			text: "Show notification?",
+			default: true,
+			value: true,
+			desktop_only: true
+		}
+	},
+	
+	
+	run: function() {
```
Comment (invalidCards) [inline Extensions/permalink_copy.js:109]: Move this to the top as well, under the preferences, so that the internal functions are between the `run` and `destroy` functions for consistency

File: Extensions/permalink_copy.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+			document.addEventListener('click', onPermalinkClick);
+			document.addEventListener('auxclick', onPermalinkClick);
+		} else if (document.attachEvent) {
+			document.attachEvent('onclick', onPermalinkClick);
+			document.attachEvent('onauxclick ', onPermalinkClick);
+		} 
+	},
+	
+	destroy: function() {
+		this.running = false;
```
Comment (invalidCards) [inline Extensions/permalink_copy.js:121]: Remove the listeners that are set by the `run` function here as well

**Review by invalidCards — CHANGES_REQUESTED** [review]

File: Extensions/permalink_copy.js (hunk 3 above)
Comment (manavortex) [reply Extensions/permalink_copy.js:4]: Too much copy-paste, will add that to something actually meaningful.

File: Extensions/permalink_copy.js (hunk 5 above)
Comment (manavortex) [reply Extensions/permalink_copy.js]: Can you rephrase? I didn't register any object functions to save two stackframes. Want them inside of it?

File: Extensions/permalink_copy.js (hunk 4 above)
Comment (manavortex) [reply Extensions/permalink_copy.js]: I moved the object definition, I hope that was what you meant. The linter wouldn't let me move the preferences outside of the class.

File: Extensions/permalink_copy.js (hunk 5 above)
Comment (invalidCards) [reply Extensions/permalink_copy.js]: Oh sorry, I misread the code a bit. Like this is fine to me :stuck_out_tongue:

**Comment by invalidCards** [comment]
By the way, you can safely remove the empty CSS file, since your extension isn't using it anyway.

**Comment by AprilSylph** [comment]
sorry, but i'm not interested in reviewing this. i don't see the need for a script that changes the default behaviour of all `:link` elements, especially on tumblr, where posts have their own "Copy link" button already. all this does is save a single right-click and cursor move to the "Copy Link Location" item.

**Comment by manavortex** [comment]
> sorry, but i'm not interested in reviewing this. i don't see the need for a script that changes the default behaviour of all `:link` elements, especially on tumblr, where posts have their own "Copy link" button already. all this does is save a single right-click and cursor move to the "Copy Link Location" item on legacy pages.

It's not changing the behaviour for all links - I just hooked all links to avoid browser compatibility issues. If you have a preferred way to hook that, let me know? 

I have a browser script for myself and figured that I might share it after two of my friends started screaming when they heard it, that's all. ¯\_(ツ)_/¯

**Comment by invalidCards** [comment]
It does raise another question from my side though - what kind of links would have the class "permalink"? There are links on the new dashboard with an `aria-label` of Permalink, but that obviously doesn't work for this extension, and all other classes on the new dashboard are auto-generated. What exactly is the use case for this?

**Comment by manavortex** [comment]
> It does raise another question from my side though - what kind of links would have the class "permalink"? There are links on the new dashboard with an `aria-label` of Permalink, but that obviously doesn't work for this extension, and all other classes on the new dashboard are auto-generated. What exactly is the use case for this?

I found those on three different blogs - didn't realize they were theme-specific, how are the chances? Damn.
Will hook up further scripting for a better selector, then. Maybe regex on href? 

If you have any ideas how best to hook this, I'd love to hear.

**Comment by invalidCards** [comment]
I guess I'm mainly missing the point of what exactly this extension is meaning to accomplish. As April said, you can just right-click links and copy the location - what is the problem that this extension is trying to solve?

**Comment by manavortex** [comment]
> I guess I'm mainly missing the point of what exactly this extension is meaning to accomplish. As April said, you can just right-click links and copy the location - what is the problem that this extension is trying to solve?

Me being lazy af and not wanting to invest the additional two clicks every single time.
I know for a fact that I'm not the only person who feels that way, but most of the others I've talked to aren't programmers, so I figured I'd socialize a hack while looking into how xkit plugin development works. I'm aware of the fact that it's an edge use case and mostly convenience, hence why it's optional.
